src/Fixer/FunctionNotation/PhpdocToParamTypeFixer.php

```diff
@@ -0,0 +1,458 @@
+     * Determine whether the function already has a param type hint.
+     *
+     * @param Tokens $tokens
+     * @param int    $index  The index of the end of the function definition line, EG at { or ;
+     *
+     * @return bool
+     */
+    private function hasParamTypeHint(Tokens $tokens, $index)
+    {
+        return $tokens[$index]->isGivenKind([T_STRING, T_NS_SEPARATOR, CT::T_ARRAY_TYPEHINT, T_CALLABLE, CT::T_NULLABLE_TYPE]);
+    }
+
+    /**
+     * @param string $paramType
+     * @param Tokens $tokens
+     * @param int    $index       The index of the end of the function definition line, EG at { or ;
+     * @param bool   $hasNull
+     * @param bool   $hasArray
+     * @param bool   $hasIterable
+     * @param bool   $hasVoid
+     * @param bool   $hasString
+     * @param bool   $hasInt
+     * @param bool   $hasFloat
+     * @param bool   $hasBool
+     * @param bool   $hasCallable
+     * @param bool   $hasObject
+     */
+    private function fixFunctionDefinition(
+        $paramType,
+        Tokens $tokens,
+        $index,
+        $hasNull,
+        $hasArray,
+        $hasIterable,
+        $hasVoid,
+        $hasString,
+        $hasInt,
+        $hasFloat,
+        $hasBool,
+        $hasCallable,
+        $hasObject
+    ) {
+        if (true === $hasNull) {
+            $newTokens[] = new Token([CT::T_NULLABLE_TYPE, '?']);
+        }
+        if (true === $hasVoid) {
+            $newTokens[] = new Token('void');
+        }
+        if (true === $hasIterable && true === $hasArray) {
+            $newTokens[] = new Token([CT::T_ARRAY_TYPEHINT, 'array']);
+        } elseif (true === $hasIterable) {
+            $newTokens[] = new Token([T_STRING, 'iterable']);
+        } elseif (true === $hasArray) {
+            $newTokens[] = new Token([CT::T_ARRAY_TYPEHINT, 'array']);
+        }
+        if (true === $hasString) {
+            $newTokens[] = new Token([T_STRING, 'string']);
+        }
+        if (true === $hasInt) {
+            $newTokens[] = new Token([T_STRING, 'int']);
+        }
+        if (true === $hasFloat) {
+            $newTokens[] = new Token([T_STRING, 'float']);
+        }
+        if (true === $hasBool) {
+            $newTokens[] = new Token([T_STRING, 'bool']);
+        }
+        if (true === $hasCallable) {
+            $newTokens[] = new Token([T_CALLABLE, 'callable']);
+        }
+        if (true === $hasObject) {
+            $newTokens[] = new Token([T_STRING, 'object']);
+        }
+
+        foreach (explode('\\', $paramType) as $nsIndex => $value) {
+            if (0 === $nsIndex && '' === $value) {
+                continue;
+            }
+
+            if (0 < $nsIndex) {
+                $newTokens[] = new Token([T_NS_SEPARATOR, '\\']);
+            }
+            $newTokens[] = new Token([T_STRING, $value]);
+        }
+        $newTokens[] = new Token([T_WHITESPACE, ' ']);
+        $tokens->insertAt($index, $newTokens);
+    }
+
+    private function typeIsOfIterable($type)
```

Comment:
I think those `typeIs*()` functions are superfluous and I would drop them. The comparaisons can be done directly where needed.

Also, the result of a `===` expression is already a boolean, no need to use a ternary.